compiler/rustc_resolve/src/ident.rs

```diff
@@ -968,7 +968,8 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
             ) {
                 Err(Determined) => continue,
                 Ok(binding)
-                    if !self.is_accessible_from(binding.vis, single_import.parent_scope.module) =>
+                    if !self.is_accessible_from(binding.vis, single_import.parent_scope.module)
+                        || binding.is_dummy_binding_import() =>
```

Comment:
So it's suspicious that we are going through `resolution.single_imports` at all at this stage.

When resolution of an import is determinate, that import is removed from `single_imports` (see `fn resolve_import`).

Finalizing an import as unresolved and setting dummy bindings for it should lead to the same outcome, I think.

Reply:
> resolution of an import is determinate

During the execution of `resolve_imports`, [resolve_ident_in_module](https://github.com/rust-lang/rust/blob/master/compiler/rustc_resolve/src/imports.rs#L671) returns `Err(Undetermined)` instead of `Err(Determined)`, so it will not remove single_import